superdesk/activity.py

```diff
@@ -242,8 +244,8 @@ def notify_and_add_activity(activity_name, msg, resource=None, item=None, user_l
         **data,
     )
 
-    if activity_name == ACTIVITY_ERROR or user_list:
-        recipients = get_recipients(user_list, activity_name)
+    if activity_name == ACTIVITY_ERROR or users_with_enabled_email_notifications:
+        recipients = get_recipients(users_with_enabled_email_notifications, activity_name)
```

Comment:
actually I'm wondering why we need to pass users with enabled notifications if it checks that in the `get_recipients` method?

Reply:
In the `get_recipients` method, we check `email_notifications`, and in the `users_with_enabled_email_notifications`, we have users whose `mark_for_user` notifications are enabled, both preferences are different.

Reply:
I see, it would make sense then probably to have this code somehow generic and do the preferences based on activity name, so we don't have to implement some extra code for pre-filtering users each time we want to have it configurable. so that `get_recipients` method would look at `f'email_notification_{activity_name}'` preference

Reply:
got it, Thanks